catroid/src/androidTest/java/org/catrobat/catroid/test/cloudmessaging/CustomFirebaseMessagingServiceTest.java

```diff
@@ -0,0 +1,126 @@
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.util.HashMap;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class CustomFirebaseMessagingServiceTest {
+
+	private NotificationBuilderProvider mockNotificationBuilderProvider;
+	private NotificationCompat.Builder mockNotificationBuilder;
+	private Notification mockNotification;
+	private NotificationManager mockNotificationManager;
+
+	@Before
+	public void setUp() {
+		mockNotificationBuilder = Mockito.mock(NotificationCompat.Builder.class, Mockito.RETURNS_SELF);
+		mockNotification = Mockito.mock(Notification.class);
+		mockNotificationBuilderProvider = Mockito.mock(NotificationBuilderProvider.class);
+		when(mockNotificationBuilderProvider.get()).thenReturn(mockNotificationBuilder);
+		when(mockNotificationBuilder.build()).thenReturn(mockNotification);
+		mockNotificationManager = Mockito.mock(NotificationManager.class);
+	}
+
+	@Test
+	public void testNotificationIsBuildAndShown() {
+		HashMap<String, String> data = new HashMap<>();
+		data.put(CloudMessage.TITLE, "Title");
+		data.put(CloudMessage.MESSAGE, "Message");
+		data.put(CloudMessage.WEB_PAGE_URL, "http://TEST.COM");
+
+		Context context = InstrumentationRegistry.getTargetContext();
+		CloudMessage cloudMessage = new CloudMessage(data);
+
+		CloudMessaging cloudMessaging = new CloudMessaging(context, mockNotificationBuilderProvider, mockNotificationManager);
+		cloudMessaging.showNotification(cloudMessage);
+
+		verify(mockNotificationBuilder).setContentTitle(eq("Title"));
+		verify(mockNotificationBuilder).setContentText(eq("Message"));
+		verify(mockNotificationBuilder).setContentIntent(any(PendingIntent.class));
+		verify(mockNotificationManager).notify(anyInt(), eq(mockNotification));
+	}
+
+	@Test
+	public void testNotificationIsNotShownOnInValidUrl() {
+		HashMap<String, String> data = new HashMap<>();
+		data.put(CloudMessage.TITLE, "Title");
+		data.put(CloudMessage.MESSAGE, "Message");
+		data.put(CloudMessage.WEB_PAGE_URL, "Invalid Url");
+
+		commonTestForInValidData(data);
+	}
+
+	@Test
+	public void testNotificationIsNotShownOnRequiredFieldEmpty() {
+		HashMap<String, String> data = new HashMap<>();
+		data.put(CloudMessage.TITLE, "");
+		data.put(CloudMessage.MESSAGE, "Message");
+		data.put(CloudMessage.WEB_PAGE_URL, "http://TEST.COM");
+
+		commonTestForInValidData(data);
+	}
+
+	@Test
+	public void testNotificationIsNotShownOnRequiredFieldMissing() {
+		HashMap<String, String> data = new HashMap<>();
+		data.put(CloudMessage.MESSAGE, "Message");
+		data.put(CloudMessage.WEB_PAGE_URL, "http://TEST.COM");
+
+		commonTestForInValidData(data);
+	}
+
+	private void commonTestForInValidData(HashMap<String, String> data) {
```

Comment:
What do you think about `checkNotificationNotShownForInvalidData` or so?
btw, In**v**alid